### ISSUE TITLE
AutoLogin Indefinite loop

### ISSUE THREAD

**Issue by johnfabian** [damienbod/angular-auth-oidc-client#1226]
I'm using the AutoLogin guard and works as expected when the login is Ok but when a Forbid is return it goes into an endless. Is there something I'm missing or some event I can subscribe to so I can intercept the forbid?

Also, when I refresh the page the isAuthenticated is false. 

 this.oidcSecurityService.isAuthenticated$.subscribe((isAuthenticatedResult: boolean)=>{
     
      this.isAuthenticated = isAuthenticatedResult  //false when page is refreshed
      console.log("auth", isAuthenticatedResult);      
    });   

Is there any updated documentation anywhere?  The current documentation is out of date and hard to follow this library since a it seems a bunch has been renamed or moved into other services.

**Comment by FabianGosebrink** [comment]
Hey thanks for this. The current documentation is not outdated, it is just referring to V12 of the library and it is up to date. We also have two auto login examples which should work. 

Which version of the lib are you using? 

Thanks

Fabian

**Comment by johnfabian** [comment]
Oh, I just realized I'm using version 11. I have it working now, but Im still trying to understand the checkAuth method, I have the checkAuth method in the App Component . I"m using Client Certificates so there's no button click so maybe that is my issue but I added the follow since when I refresh the browser it returns false but everything seems to be working now with the method below, I just wrapped the isAuthentiacted

handleOAuthCallBack(): Observable<boolean> {
    return new Observable((observer) => {
      this.oidcSecurityService.isAuthenticated$.subscribe(
        (isAuthenticated: boolean) => {
                   
            this.isAuthenticatedSubject.next(isAuthenticated);
            observer.next(isAuthenticated);
            observer.complete();        
        },
        (error: any) => {
          this.isAuthenticatedSubject.next(false);
          observer.next(false);
          observer.complete();
        }
      );
    });

I created my own guard and added the following to prevent an endless loop but I'm thinking it is because of using smart cards / client certificates but it is working now

canActivate(activatedRouteSnapshot: ActivatedRouteSnapshot, state: RouterStateSnapshot): Observable<boolean> | boolean {
    //if user already has a valid token
    if (this.authService.hasValidToken()) {
      return true;
    }
    //save the url for when returning from the callback, the menu will take care of the routing after it's loaded
    this.storageService.requestedUrl = state.url;

    return new Observable((observer) => {
      this.authService.isAuthenticated$.subscribe((isAuth: boolean) => {
        if (isAuth) {
          observer.next(true);
          observer.complete();
        } 
        else if(!isAuth && this.storageService.loginAttempted !==true) {
          //auto login if not already attempted
       this.storageService.loginAttempted = true;

      this.oidcSecurityService.authorize();

          observer.next(false);
          observer.complete();
        }
        else{      
          //login attemplated and not authorized    


          observer.next(false);
          observer.complete();
        }
      });
    });
    
  }

Thanks for the the quick reply!

**Comment by FabianGosebrink** [comment]
Hey, so if you can, please update to V12. We fixed some bugs in the Auto Login field there [Auto Login](https://nice-hill-002425310.azurestaticapps.net/docs/documentation/features#auto-login)

The `checkAuth()` method is called when you return from the STS with a code in the URL. It processes the code in the URL and sets the values. If you do not have any info in the URL it checks the state of the lib and returns if you're authenticated or not.

**Comment by johnfabian** [comment]
Ahh ok, thank you so much for all your help, I really appreciate it!
